Frame.cc

```diff
@@ -437,7 +435,9 @@ bool Frame::SetRegionSpectralRequirements(int region_id, const std::vector<CARTA
     if (_regions.count(region_id)) {
         auto& region = _regions[region_id];
         region_ok = region->SetSpectralRequirements(profiles, NumStokes());
-        SetRegionSpectralRequests(region_id, profiles);
+        if (profiles.size() > 0) {
+            SetRegionId(region_id);
```

Comment:
There can be multiple regions with spectral profiles requested if the user opens additional zprofile widgets, so saving the latest region id and checking it will not work for interrupting a spectral profile.  Stop only if this profile's parameters change (its region changed, stokes changed, requirements changed).